### DISCUSSION TITLE
How to set the CreatedBy (and ModifiedBy) of records

### DISCUSSION THREAD

**Discussion by bwmodular** [DynamicsValue/fake-xrm-easy#98] [Q&A] [answered]
I have a plugin which needs to look at the details of the user who originally created the record.
How can I set the CreatedBy on a record created in FakeXrmEasy to be anything other than the default user? CreatedBy is a readonly attribute, so I can't set that when I create the record.

In the same plugin I need to get details of the user who is performing the update. I can't work out how to do this either. I tried changing the callerid of the service but that doesn't seem possible.

Sorry if I'm missing something obvious...

**Comment by jordimontana82** [comment, marked as answer]
Hi Ben!

All read-only properties could be set using the brackets notation [] even if they're read only.

Here's a sample of CreatedBy.

https://github.com/DynamicsValue/fake-xrm-easy-core/blob/f343198c49c5eb68b1940173c7668b2505cf5b75/tests/FakeXrmEasy.Core.Tests/Query/QueryLookupTests/QueryLookupTests.cs#L71

Hopefully that will work!

**Reply by jordimontana82** [reply]
@bwmodular did that work?

**Comment by bwmodular** [comment]
Thanks @jordimontana82 - works a treat!
